### PR TITLE
[apptools]Add apptools test suite for linux

### PR DESCRIPTION
**Pull request by yinhaichao**
Impacted tests(approved): new 18, update 0, delete 0
Unit test platform: [Linux]
Unit test result summary: pass 16, fail 2, block 0

Fail reason:
  - Create pkgname like "org.example.1234test/org.example.1234" successfully, but expect failed

Bug id: xwalk-3818

### CODE REVIEW

**Comment by ibelem** [comment]
@yinhaichao @haoxli The XWALK-3682 is feature ID not bug id, could you please file bugs for the failed issues?

- In the create.py case of mistake is because the directory already exists, why not delete the directory before creating in automation scripts?

File: apptools/apptools-linux-tests/README [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+This test suite is for wrt-apptools-linux-tests
+1. ./allpairs.py is create pkgName.py and report directory for package name
+
+## Precondition
+
+1. The Node.js, the debuild tool, the Crosswalk runtime, and git must be functional
+2. Download crosswalk-app-tools and configure it
+    2.1 Download Crosswalk app tools:  git clone https://github.com/crosswalk-project/crosswalk-app-tools.git  
+    2.2 Initialize the Crosswalk app tools:  cd crosswalk-app-tools , then  npm install  
+    3.3 Checkout the iOS backend:  cd node_modules , then  git clone https://github.com/crosswalk-project/crosswalk-app-tools-deb.git crosswalk-app-tools-backend-deb
```
Comment (Honry) [inline apptools/apptools-linux-tests/README]: iOS backend? It must be Deepin backend

File: apptools/apptools-linux-tests/README [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    3.3 Checkout the iOS backend:  cd node_modules , then  git clone https://github.com/crosswalk-project/crosswalk-app-tools-deb.git crosswalk-app-tools-backend-deb  
+    4.4 Install dependencies:  cd crosswalk-app-tools-backend-deb , then  npm install , and  cd ../..  
+    5.5 The main script is  crosswalk-app-tools/bin/crosswalk-app . Set environment PATH or invoke with directory
+
+## Test Steps
+
+1. unzip wrt-apptools-linux-tests<version>.zip -d [testprefix-path]
+2. cd [testprefix-path]/opt/wrt-apptools-linux-tests/
+3. run test case
+   testkit-lite -f ....
```
Comment (Honry) [inline apptools/apptools-linux-tests/README]: Why omit testkit-lite execution command? Does testkit-lite support deepin now?

**Comment by Honry** [comment]
And what is apptools/apptools-linux-tests/tempdata/.temp used for?

File: apptools/apptools-linux-tests/tempdata/README [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## Introduction
```
Comment (Honry) [inline apptools/apptools-linux-tests/tempdata/README]: Seems this folder is useless, please remove it.

File: apptools/apptools-linux-tests/tests.full.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="./testcase.xsl"?>
+<test_definition>
+  <suite category="Crosswalk App Tools" name="wrt-apptools-linux-tests">
+    <set name="apptools" type="pyunit">
+      <testcase component="Crosswalk App Tools/Project Create" execution_type="auto" id="Crosswalk_create" priority="P1" purpose="Validate if App Tool can work well with create option" status="approved" type="Functional" subcase="3">
```
Comment (Honry) [inline apptools/apptools-linux-tests/tests.full.xml]: Please update component name with "Crosswalk App Tools/CLI", and add prefix "Linux - "before purpose to distinguish different platforms.

File: apptools/apptools-linux-tests/README [hunk 5]
```diff
@@ -0,0 +1,31 @@
+## Introduction
+
+This test suite is for wrt-apptools-linux-tests
```
Comment (Honry) [inline apptools/apptools-linux-tests/README]: Please update the test suite name

File: apptools/apptools-linux-tests/README [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+2. Download crosswalk-app-tools and configure it
+    2.1 Download Crosswalk app tools:  git clone https://github.com/crosswalk-project/crosswalk-app-tools.git  
+    2.2 Initialize the Crosswalk app tools:  cd crosswalk-app-tools , then  npm install  
+    3.3 Checkout the deb backend:  cd node_modules , then  git clone https://github.com/crosswalk-project/crosswalk-app-tools-deb.git crosswalk-app-tools-backend-deb  
+    4.4 Install dependencies:  cd crosswalk-app-tools-backend-deb , then  npm install , and  cd ../..  
+    5.5 The main script is  crosswalk-app-tools/bin/crosswalk-app . Set environment PATH or invoke with directory
+
+## Test Steps
+
+1. unzip wrt-apptools-linux-tests<version>.zip -d [testprefix-path]
```
Comment (Honry) [inline apptools/apptools-linux-tests/README]: ditto

File: apptools/apptools-linux-tests/README [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+2. cd [testprefix-path]/opt/wrt-apptools-linux-tests/
```
Comment (Honry) [inline apptools/apptools-linux-tests/README]: ditto

File: apptools/apptools-linux-tests/tests.full.xml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        <description>
+          <test_script_entry>/opt/wrt-apptools-linux-tests/apptools/create.py</test_script_entry>
+        </description>
+      </testcase>
+      <testcase component="Crosswalk App Tools/CLI" execution_type="auto" id="Crosswalk_build" priority="P1" purpose="Linux - Validate if App Tool can work well with build option" status="designed" type="Functional" subcase="2">
+        <description>
+          <test_script_entry>/opt/wrt-apptools-linux-tests/apptools/build.py</test_script_entry>
+        </description>
+      </testcase>
+      <testcase component="Crosswalk App Tools/Project Create" execution_type="auto" id="Crosswalk_pkgName" priority="P1" purpose="Validate if the App Tool rules about package name" status="approved" type="Functional" subcase="16">
```
Comment (Honry) [inline apptools/apptools-linux-tests/tests.full.xml]: Miss one to add 'Linux -' prefix in purpose.

**Pull request merged**